src/bidiMapper/modules/cdp/CdpTarget.ts

```diff
@@ -169,6 +186,36 @@ export class CdpTarget {
     });
   }
 
+  #restoreFrameTreeState(frameTree: Protocol.Page.FrameTree) {
+    const frame = frameTree.frame;
+    if (this.#browsingContextStorage.findContext(frame.id) === undefined) {
+      if (frame.parentId === undefined) {
+        // TODO: do not throw.
+        throw new UnknownErrorException('Parent frame not found');
+      }
+      const parentBrowsingContext = this.#browsingContextStorage.getContext(
+        frame.parentId
+      );
+      BrowsingContextImpl.create(
+        frame.id,
+        frame.parentId,
+        parentBrowsingContext.userContext,
+        parentBrowsingContext.cdpTarget,
+        this.#eventManager,
+        this.#browsingContextStorage,
+        this.#realmStorage,
+        // At this point, we don't know the URL of the frame yet, so it will be updated
+        // later.
+        frame.url,
+        this.#logger,
+        false
```

Comment:
I believe we still have to emit `browsingContext.contextCreated` if the target is attached with frames.

Reply:
I think that should be only on subscribe or when a new context actually created? anyway the tests didn't look happy with extra events